File: server/router/mine/mine_blogs.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineBlogsRouter struct{}

func (s *MineBlogsRouter) InitMineBlogsRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineBlogsRouter := Router.Group("mineBlogs").Use(middleware.OperationRecord())
	mineBlogsRouterWithoutRecord := Router.Group("mineBlogs")
	mineBlogsRouterWithoutAuth := PublicRouter.Group("mineBlogs")
	{
		mineBlogsRouter.POST("createMineBlogs", mineBlogsApi.CreateMineBlogs)
		mineBlogsRouter.DELETE("deleteMineBlogs", mineBlogsApi.DeleteMineBlogs)
		mineBlogsRouter.DELETE("deleteMineBlogsByIds", mineBlogsApi.DeleteMineBlogsByIds)
		mineBlogsRouter.PUT("updateMineBlogs", mineBlogsApi.UpdateMineBlogs)
	}
	{
		mineBlogsRouterWithoutRecord.GET("findMineBlogs", mineBlogsApi.FindMineBlogs)
		mineBlogsRouterWithoutRecord.GET("getMineBlogsList", mineBlogsApi.GetMineBlogsList)
		mineBlogsRouterWithoutRecord.GET("getMineBlogsStat", mineBlogsApi.GetMineBlogsStat)
	}
	{
		_ = mineBlogsRouterWithoutAuth
	}
}

File: web/src/api/mine/mine_blogs.js
import service from '@/utils/request'

export const createMineBlogs = (data) => {
  return service({
    url: '/mineBlogs/createMineBlogs',
    method: 'post',
    data
  })
}

export const deleteMineBlogs = (params) => {
  return service({
    url: '/mineBlogs/deleteMineBlogs',
    method: 'delete',
    params
  })
}

export const deleteMineBlogsByIds = (params) => {
  return service({
    url: '/mineBlogs/deleteMineBlogsByIds',
    method: 'delete',
    params
  })
}

export const updateMineBlogs = (data) => {
  return service({
    url: '/mineBlogs/updateMineBlogs',
    method: 'put',
    data
  })
}

export const findMineBlogs = (params) => {
  return service({
    url: '/mineBlogs/findMineBlogs',
    method: 'get',
    params
  })
}

export const getMineBlogsList = (params) => {
  return service({
    url: '/mineBlogs/getMineBlogsList',
    method: 'get',
    params
  })
}

export const getMineBlogsStat = () => {
  return service({
    url: '/mineBlogs/getMineBlogsStat',
    method: 'get'
  })
}
